### PR TITLE
[Merged by Bors] - chore(Data/Fin/Tuple/Basic): Fill in the API hole (partly)

### PR DESCRIPTION
**Pull request by YaelDillies**
As explained by the module doc, we are missing the `Fin.succAbove` equivalent of `Fin.tail`/`Fin.init`. In fact, it already exists as the second projection of `extractNth`, so this PR deletes `extractNth` and defines its second projection as a new function `removeNth`.

---

- [x] depends on: #13567

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Fin/Tuple/Basic.lean [hunk 1]
```diff
@@ -65,7 +65,9 @@ For a **pivot** `p : Fin (n + 1)`,
   `Fin.insertNth f a : Fin (n + 1) → α` by adding `a` in position `p`. In general, tuples can be
   dependent functions, in which case `f : ∀ i : Fin n, α (p.succAbove i)` and `a : α p`. This is a
   special case of `Fin.succAboveCases`.
-* **There is currently no equivalent of `Fin.tail`/`Fin.init` for adding in the middle.**
+* `Fin.removeNth`: Turn a tuple `f : Fin (n + 1) → α` into a tuple `Fin.removeNth p f : Fin n → α`
+  by forgetting the start. In general, tuples can be dependent functions,
```
Comment (b-mehta) [inline Mathlib/Data/Fin/Tuple/Basic.lean:69]: It isn't forgetting the start, it's forgetting p

File: Mathlib/Data/Fin/Tuple/Basic.lean [hunk 2]
```diff
@@ -1016,19 +1030,14 @@ theorem preimage_insertNth_Icc_of_not_mem {i : Fin (n + 1)} {x : α i} {q₁ q
 
 /-- Separates an `n+1`-tuple, returning a selected index and then the rest of the tuple.
 Functional form of `Equiv.piFinSuccAbove`. -/
+@[deprecated removeNth (since := "2024-06-19")]
 def extractNth (i : Fin (n + 1)) (f : (∀ j, α j)) :
     α i × ∀ j, α (i.succAbove j) :=
   (f i, fun j => f (i.succAbove j))
```
Comment (eric-wieser) [inline Mathlib/Data/Fin/Tuple/Basic.lean:1036]:
```suggestion
  (f i, removeNth f i)
```
or similar

File: Mathlib/Data/Fin/Tuple/Basic.lean [hunk 3]
```diff
@@ -1016,19 +1030,14 @@ theorem preimage_insertNth_Icc_of_not_mem {i : Fin (n + 1)} {x : α i} {q₁ q
...
-theorem extractNth_insertNth {i : Fin (n + 1)} (x : α i) (p : ∀ j : Fin n, α (i.succAbove j)) :
-    i.extractNth (i.insertNth x p) = (x, p) := by
-  simp [extractNth]
-
-@[simp]
-theorem insertNth_extractNth {i : Fin (n + 1)} (f : ∀ j, α j) :
-    i.insertNth (i.extractNth f).1 (i.extractNth f).2 = f := by
-  simp [Fin.extractNth, Fin.insertNth_eq_iff]
+theorem insertNth_self_removeNth (p : Fin (n + 1)) (f : ∀ j, α j) :
+    insertNth p (f p) (removeNth p f) = f := by simp [Fin.insertNth_eq_iff]
```
Comment (eric-wieser) [inline Mathlib/Data/Fin/Tuple/Basic.lean:1040]: Can you add the more general statement that `insertNth p x (removeNth p f) = Function.update f p x`?

**Comment by eric-wieser** [comment]
bors merge

Thanks!

**Pull request closed without merging**